packages/provider.mustache/src/AttributeMustacheProvider.ts

```diff
@@ -26,65 +30,67 @@ const DEFAULT_ATTRIBUTE_BINDING_MAP = {
  *  Interpret {{ }} inside DOM attributes e.g. <div class='{{ classes }}'>
  */
 export default class AttributeMustacheProvider extends Provider {
+  ATTRIBUTES_TO_SKIP: Set<string>
+  ATTRIBUTES_BINDING_MAP: any
   get FOR_NODE_TYPES () { return [ 1 ] } // document.ELEMENT_NODE
 
-  constructor (params = {}) {
+  constructor (params:ProviderParamsInput|null = null) {
     super(params)
-    this.ATTRIBUTES_TO_SKIP = new Set(params.attributesToSkip || ['data-bind'])
-    this.ATTRIBUTES_BINDING_MAP = params.attributesBindingMap || DEFAULT_ATTRIBUTE_BINDING_MAP
+    this.ATTRIBUTES_TO_SKIP = new Set(params?.attributesToSkip || ['data-bind'])
+    this.ATTRIBUTES_BINDING_MAP = params?.attributesBindingMap || DEFAULT_ATTRIBUTE_BINDING_MAP
   }
 
-  * attributesToInterpolate (attributes) {
+  * attributesToInterpolate (attributes: NamedNodeMap) {
     for (const attr of Array.from(attributes)) {
       if (this.ATTRIBUTES_TO_SKIP.has(attr.name)) { continue }
       if (attr.specified && attr.value.includes('{{')) { yield attr }
     }
   }
 
-  nodeHasBindings (node) {
+  nodeHasBindings (node: Element) {
     return !this.attributesToInterpolate(node.attributes).next().done
   }
 
-  partsTogether (parts, context, node, ...valueToWrite) {
+  partsTogether (parts: any[], context: any, node: Element, ...valueToWrite: any[]) {
     if (parts.length > 1) {
       return parts
-        .map(p => unwrap(p.asAttr(context, this.globals, node))).join('')
+        .map((p: { asAttr: (arg0: any, arg1: any, arg2: Element) => any }) => unwrap(p.asAttr(context, this.globals, node))).join('')
     }
     // It may be a writeable observable e.g. value="{{ value }}".
     const part = parts[0].asAttr(context, this.globals)
     if (valueToWrite.length) { part(valueToWrite[0]) }
     return part
   }
 
-  attributeBinding (name, parts) {
+  attributeBinding (name: string, parts: any[]): (string | any[])[] {
```

Comment:
can change this to `...): [string, any[]]`, for strong tuple type

Reply:
Done, see https://github.com/Auge19/tko/commit/cd93508d0ebb1ad1251168a247f3c147f2748103